a10_octavia/controller/worker/tasks/a10_database_tasks.py

```diff
@@ -734,3 +735,67 @@ def execute(self, lb_resource):
                     id=flavor.flavor_profile_id)
                 flavor_data = json.loads(flavor_profile.flavor_data)
                 return self._format_keys(flavor_data)
+
+
+class GetNatPoolEntry(BaseDatabaseTask):
+
+    def execute(self, member, nat_flavor=None):
+        if nat_flavor and 'pool_name' in nat_flavor:
+            return self.nat_pool_repo.get(db_apis.get_session(), name=nat_flavor['pool_name'],
+                                          subnet_id=member.subnet_id)
+
+
+class UpdateNatPoolDB(BaseDatabaseTask):
+
+    def execute(self, member, nat_flavor=None, nat_pool=None, subnet_port=None):
+        if nat_flavor is None:
+            return
+
+        if nat_pool is None:
+            if subnet_port is None:
+                raise exceptions.PortIdMissing()
+
+            id = uuidutils.generate_uuid()
+            pool_name = nat_flavor['pool_name']
```

Comment:
In order to ensure there is no key error, we can do `nat_flavor.get('pool_name')` which returns None, if not present ---> if `nat_flavor` is  a dict

Similar is case for others - `nat_flavor['start_address']`, `nat_flavor['end_address']`

Reply:
ok